packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -12,6 +12,9 @@ import 'print.dart';
 /// Signature for [FlutterError.onError] handler.
 typedef FlutterExceptionHandler = void Function(FlutterErrorDetails details);
 
+/// Signature for [DiagnosticPropertiesBuilder] transformer.
+typedef DiagnosticPropertiesTransformer = List<DiagnosticsNode> Function(List<DiagnosticsNode> properties);
```

Comment:
might be worth making this Iterator -> Iterator rather than List -> List.

Reply:
it is probably not worth it as the given property is a list and the output will be assign back to a list. it might add some unnecessary .tolist or list.from() conversion